plugins/pulp_docker/plugins/importers/sync.py

```diff
@@ -218,91 +203,59 @@ def find_and_read_ancestry_file(image_id, parent_dir):
             return json.load(ancestry_file)
 
 
-class GetLocalImagesStep(GetLocalUnitsStep):
-    def _dict_to_unit(self, unit_dict):
-        """
-        convert a unit dictionary (a flat dict that has all unit key, metadata,
-        etc. keys at the root level) into a Unit object. This requires knowing
-        not just what fields are part of the unit key, but also how to derive
-        the storage path.
-
-        Any keys in the "metadata" dict on the returned unit will overwrite the
-        corresponding values that are currently saved in the unit's metadata. In
-        this case, we pass an empty dict, because we don't want to make changes.
-
-        :param unit_dict:   a flat dictionary that has all unit key, metadata,
-                            etc. keys at the root level, representing a unit
-                            in pulp
-        :type  unit_dict:   dict
-
-        :return:    a unit instance
-        :rtype:     pulp.plugins.model.Unit
-        """
-        model = DockerImage(unit_dict['image_id'], unit_dict.get('parent_id'),
-                            unit_dict.get('size'))
-        return self.get_conduit().init_unit(model.TYPE_ID, model.unit_key, {},
-                                            model.relative_path)
+class SaveDockerUnits(SaveUnitsStep):
 
-
-class SaveUnits(PluginStep):
-    def __init__(self, working_dir):
-        """
-        :param working_dir: full path to the directory into which image files
-                            are downloaded. This directory should contain one
-                            directory for each docker image, with the ID of the
-                            docker image as its name.
-        :type  working_dir: basestring
-        """
-        super(SaveUnits, self).__init__(step_type=constants.SYNC_STEP_SAVE,
-                                        plugin_type=constants.IMPORTER_TYPE_ID,
-                                        working_dir=working_dir)
+    def __init__(self, step_type=constants.SYNC_STEP_SAVE):
+        super(SaveDockerUnits, self).__init__(step_type=step_type)
         self.description = _('Saving images and tags')
 
-    def process_main(self):
+    def get_iterator(self):
+        # Return a generator so that the platform will show a returned value
+        # if this returns an empty list the platform will call process_main once with item=none
+        def generator():
+            for item in self.parent.step_get_local_units.units_to_download:
+                yield item
+        return generator()
```

Comment:
You might be able to just do:

``return iter(self.parent.step_get_local_units.units_to_download)``

or if not, then a generator expression:

``return (i for i in self.parent.step_get_local_units.units_to_download)``

Either would simplify this method.